src/relations/backend_database.py

```diff
@@ -100,6 +105,9 @@ def _on_relation_broken(self, event: RelationBrokenEvent):
             cfg = self.charm.read_pgb_config()
             user = self.charm.backend_postgres.user
             self.charm.remove_user(user, cfg=cfg, reload_pgbouncer=True, render_cfg=True)
+
+            self.charm.update_postgres_endpoints()
```

Comment:
This is not updating (removing) the PostgreSQL endpoints in the pgbouncer config file. As the endpoints are still present in the databag during this event, they are kept in the pgbouncer config file.